rust/operator-binary/src/controller.rs

```diff
@@ -905,11 +910,12 @@ async fn build_node_rolegroup_statefulset(
         .metadata_builder(|m| {
             m.with_recommended_labels(build_recommended_labels(
                 nifi,
-                image_version,
+                &resolved_product_image.app_version_label,
                 &rolegroup_ref.role,
                 &rolegroup_ref.role_group,
             ))
         })
+        .image_pull_secrets_from_product_image(resolved_product_image)
```

Comment:
You've set the pull secrets on the monitoring job as well? Changing the image will be done after merging the tools image

Reply:
Agreed. But IMHO we should set the pullSecrets already (it will not destroy anything), so they don't get missed when people simply swap out the tools image.

Reply:
done